src/API/NoteActions.php

```diff
@@ -107,6 +107,11 @@ public function trigger_note_action( $request ) {
 			$note->set_status( $triggered_action->status );
 		}
 
+		// Update note state depending on its type.
+		if ( 'survey' === $note->get_type() ) {
```

Comment:
This checking of type and performing some type-specific logic here is not ideal. Maybe you could hook into this using the `woocommerce_note_action` action instead, so that you could keep this deleting logic in `WC_Admin_Notes_Insight_First_Sale.php`.

Reply:
I see, it's a nice catch @mattsherman! This gives me an idea😀
What do you think about creating a hook and a deletion method [in WC_Admin_Notes.php](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Notes/WC_Admin_Notes.php#L25) for this type of note?
Doing that I could follow the same pattern we use to [delete the marketing notes](https://github.com/woocommerce/woocommerce-admin/blob/main/src/Notes/WC_Admin_Notes.php#L227) and we don't have to hook the deletion method in every note of this type.

Reply:
@octaedro I think I follow your idea. I think that would be good.